### PR TITLE
Slight tweak in azure SB long running operations

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

**Comment by danielmarbach** [comment]
@SimonCropp @SeanFeldman Slight tweak which makes the sample a bit more "idiomatic"

File: samples/azure/azure-service-bus-long-running/ASB_7/Processor/Processor.cs [hunk 1]
```diff
@@ -13,36 +13,40 @@ public class Processor
...
+
         var cloudTableClient = StorageHelper.GetTableClient();
         table = cloudTableClient.GetTableReference(Constants.TableName);
-        table.CreateIfNotExists();
-
-        var task = new Task(() =>
-        {
-#pragma warning disable 4014
-            #region tasks
+        await table.CreateIfNotExistsAsync(cancellationToken).ConfigureAwait(false);
```
Comment (SeanFeldman) [inline samples/azure/azure-service-bus-long-running/ASB_7/Processor/Processor.cs]: For some reason async version didn't work... needs to be verified.

File: samples/azure/azure-service-bus-long-running/ASB_7/Processor/Processor.cs [hunk 2]
```diff
@@ -13,36 +13,40 @@ public class Processor
...
         endpoint = endpointInstance;
+
         var cloudTableClient = StorageHelper.GetTableClient();
         table = cloudTableClient.GetTableReference(Constants.TableName);
-        table.CreateIfNotExists();
-
-        var task = new Task(() =>
-        {
-#pragma warning disable 4014
-            #region tasks
```
Comment (SeanFeldman) [inline samples/azure/azure-service-bus-long-running/ASB_7/Processor/Processor.cs]: region is needed by the documentation

File: samples/azure/azure-service-bus-long-running/ASB_7/Processor/Processor.cs [hunk 3]
```diff
@@ -13,36 +13,40 @@ public class Processor
...
+    public async Task Stop()
+    {
+        cancellationTokenSource.Cancel();
+        await Task.WhenAll(pollingTask, processingTask).ConfigureAwait(false);
+        cancellationTokenSource.Dispose();
     }
 
     async Task StartPolling(CancellationToken token)
     {
         // should do CancellationToken and handle TaskCanceledException
```
Comment (SeanFeldman) [inline samples/azure/azure-service-bus-long-running/ASB_7/Processor/Processor.cs]: comment can be removed now as it's not needed

File: samples/azure/azure-service-bus-long-running/ASB_7/Processor/Program.cs [hunk 4]
```diff
@@ -30,15 +29,14 @@ static void Main()
         var endpointInstance = await Endpoint.Start(endpointConfiguration)
             .ConfigureAwait(false);
 
-        var cancellationSource = new CancellationTokenSource();
         var processor = new Processor();
 
-        processor.Start(endpointInstance, cancellationSource.Token);
+        await processor.Start(endpointInstance).ConfigureAwait(false);
```
Comment (SeanFeldman) [inline samples/azure/azure-service-bus-long-running/ASB_7/Processor/Program.cs]: :+1:

File: samples/azure/azure-service-bus-long-running/ASB_7/Processor/Processor.cs (hunk 2 above)
Comment (danielmarbach) [inline samples/azure/azure-service-bus-long-running/ASB_7/Processor/Processor.cs]: Missed that

**Comment by danielmarbach** [comment]
Updated

**Comment by SeanFeldman** [comment]
@Particular/docs-maintainers could you please approve? Thanks

**Pull request merged**